### PR TITLE
Devel/justin

### PR DESCRIPTION
**Pull request by greenkee**
Changed State Manager's state to a dictionary

### CODE REVIEW

File: testy/stateManager.py [hunk 1]
```diff
@@ -18,27 +18,48 @@ def __init__(self, badThingsQueue, inputQueue, runtimePipe):
...
-    self.state = [5]
+    self.state = {
+     "incrementer" : 5,
+     "int1" : 112314,
+     "float1": 987.123,
+     "bool1" : True, 
+     "dict1" : {"inner_dict1_int" : 555, "inner_dict_1_string": "hello"},
+     "list1" : [70, "five", 14.3],
+     "string1" : "abcde"
+
```
Comment (yizow) [inline testy/stateManager.py]: remove unnesscary new line

File: testy/stateManager.py [hunk 2]
```diff
@@ -18,27 +18,48 @@ def __init__(self, badThingsQueue, inputQueue, runtimePipe):
...
+     }
 
   def addPipe(self, processName, pipe):
     self.processMapping[processName] = pipe
     pipe.send(SM_COMMANDS.READY)
 
+  def getValue(self, string):
```
Comment (yizow) [inline testy/stateManager.py]: These two functions don't add much value by themselves; too short.

File: testy/stateManager.py [hunk 3]
```diff
@@ -18,27 +18,48 @@ def __init__(self, badThingsQueue, inputQueue, runtimePipe):
...
+    return self.state[string]
+
+  def setValue(self, string, value):
```
Comment (yizow) [inline testy/stateManager.py]: See above

File: testy/stateManager.py (hunk 2 above)
Comment (yizow) [inline testy/stateManager.py]: Should probably change the argument list to be "self, key"

File: testy/stateManager.py [hunk 4]
```diff
@@ -18,27 +18,48 @@ def __init__(self, badThingsQueue, inputQueue, runtimePipe):
...
 
-      elif request[0] == SM_COMMANDS.ADD:
+      elif cmd_type == SM_COMMANDS.ADD:
         self.addPipe(request[1], request[2])
-
-      elif request[0] == SM_COMMANDS.HELLO:
-        self.state[0] -= 1
-        self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(self.state[0])
+      elif cmd_type == SM_COMMANDS.GET_VAL:
+        self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(self.getValue(request[1]))
```
Comment (yizow) [inline testy/stateManager.py]: Consider moving a lot of this code into the self.getValue function

File: testy/stateManager.py [hunk 5]
```diff
@@ -18,27 +18,48 @@ def __init__(self, badThingsQueue, inputQueue, runtimePipe):
...
+      elif cmd_type == SM_COMMANDS.SET_VAL:
+        self.processMapping[PROCESS_NAMES.STUDENT_CODE].send(self.setValue(request[1], request[2]))
```
Comment (yizow) [inline testy/stateManager.py]: See above

File: testy/stateManager.py [hunk 6]
```diff
@@ -18,27 +18,48 @@ def __init__(self, badThingsQueue, inputQueue, runtimePipe):
...
+      elif cmd_type == SM_COMMANDS.HELLO:
+        print("HELLO")
+        # self.state[0] -= 1
```
Comment (yizow) [inline testy/stateManager.py]: Remove the commented out lines. Github history will track removed lines.

File: testy/studentCode.py [hunk 7]
```diff
@@ -8,9 +8,20 @@ def setup(pipe):
...
+
+  # stateQueue.put([SM_COMMANDS.GET_VAL, "float1"])
+  # response = pipe.recv()
+  # print("Get Info:", response)
+
   print("Saying hello to the other side")
-  state = Robot.getValue("")
-  print("Hello from the other side %d" % (state,))
-  1.0/state
+  print("DAT:", 1.0/response)
```
Comment (yizow) [inline testy/studentCode.py]: Add newline at end of file. 
Has semantic meaning apparently; google for more details.

**Review by yizow — CHANGES_REQUESTED** [review]
Respond to comments, either change code according to comment, or comment back explaining why the comment is incorrect. 

When finished:
git add -u;
git commit --amend;
git push -f (make sure you're pushing to devel/justin

**Review by bodiddles — COMMENTED** [review]
Yizhe, we have to eventually change indents to 4 spaces eventually.

**Pull request merged**